tokio/src/sync/tests/semaphore_batch.rs

```diff
@@ -236,7 +236,7 @@ fn close_semaphore_notifies_permit2() {
 #[test]
 fn cancel_acquire_releases_permits() {
     let s = Semaphore::new(10);
-    let _permit1 = s.try_acquire(4).expect("uncontended try_acquire succeeds");
```

Comment:
[This changes behaviour of the test](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=81ec537823cec5bfd89903f5359ad7bd). The fact that the test still passed should be investigated more closely.

Reply:
try_acquire [returns](https://github.com/tokio-rs/tokio/blob/db0d6d75b3b65812837c8561269e55eab00a74dd/tokio/src/sync/batch_semaphore.rs#L162) Result<(), Error> so `_permit1` is binded to ().

Reply:
Oh, I see. That makes sense.